pointcloud/octree/basic.go

```diff
@@ -0,0 +1,171 @@
+package octree
+
+import (
+	"context"
+
+	"github.com/edaniels/golog"
+	"github.com/golang/geo/r3"
+	"github.com/pkg/errors"
+
+	pc "go.viam.com/rdk/pointcloud"
+)
+
+// octree data is the structure of an octree.
+type basicOctree struct {
+	ctx    context.Context
+	logger golog.Logger
+	node   basicOctreeNode
+	center r3.Vector
+	side   float64
+	meta   MetaData
+}
+
+type basicOctreeNode struct {
+	nodeType NodeType
+	tree     []*basicOctree
+	point    pc.PointAndData
+}
+
+// New creates a new basic octree with specified center, side and metadata.
+func New(ctx context.Context, center r3.Vector, sideLength float64, logger golog.Logger) (Octree, error) {
```

Comment:
According to tech spec this function should be called `NewOctree`

Reply:
Spoke to Bijan about this, because we are making a specific implementation of an Octree (basicOctree) we had the options of calling it `New`, `NewOctree` or `NewBasicOctree`. As this is currently the only implementation we decided to call it `New`. In the future if other forms of octrees get implemented this may get changed to `NewBasicOctree`